### PR TITLE
new: Support image gen2 features

### PR DESCRIPTION
**Pull request by yec-akamai**
## 📝 Description

Support the new features for image gen2 in image resource and data sources. The new features including image tagging, image replication. User can use TF to replicate an image, and gets information including replications region, status, and total size. 

## ✔️ How to Test
Unit test:
```
make unit-test PKG_NAME="linode/image"
```
```
make unit-test PKG_NAME="linode/images"
```

Set up the environment to test against alpha:
```
export LINODE_URL=...
export LINODE_API_VERSION=...
export LINODE_TOKEN=...
```

Integration tests:
```
make int-test PKG_NAME="linode/image"
```
```
make int-test PKG_NAME="linode/images"
```

TODO: image replication test

### CODE REVIEW

File: linode/image/datasource_test.go [hunk 1]
```diff
@@ -36,3 +37,39 @@ func TestAccDataSourceImage_basic(t *testing.T) {
...
+					resource.TestCheckResourceAttr(resourceName, "description", "descriptive text"),
+					resource.TestCheckResourceAttrSet(resourceName, "created"),
+					resource.TestCheckResourceAttrSet(resourceName, "created_by"),
+					resource.TestCheckResourceAttrSet(resourceName, "size"),
+					resource.TestCheckResourceAttr(resourceName, "type", "manual"),
+					resource.TestCheckResourceAttr(resourceName, "is_public", "false"),
+					resource.TestCheckResourceAttrSet(resourceName, "deprecated"),
+					resource.TestCheckResourceAttrSet(resourceName, "tags.#"),
+					resource.TestCheckResourceAttrSet(resourceName, "total_size"),
+					// resource.TestCheckResourceAttr(resourceName, "replications.#", "2"),
```
Comment (yec-akamai) [inline linode/image/datasource_test.go:70]: Uncomment this check once replication is available.

File: linode/image/resource_test.go [hunk 2]
```diff
@@ -208,6 +215,47 @@ func TestAccImage_uploadFile(t *testing.T) {
 	})
 }
 
+//
+//func TestAccImage_replicate(t *testing.T) {
```
Comment (yec-akamai) [inline linode/image/resource_test.go:219]: Uncomment this test case once replication is functional.

File: linode/images/framework_schema_datasource.go [hunk 3]
```diff
@@ -14,6 +14,8 @@ var filterConfig = frameworkfilter.Config{
 	"type":       {APIFilterable: true, TypeFunc: frameworkfilter.FilterTypeString},
 	"vendor":     {APIFilterable: true, TypeFunc: frameworkfilter.FilterTypeString},
 
+	// TODO: check if tags become API filterable
```
Comment (yec-akamai) [inline linode/images/framework_schema_datasource.go:17]: The `tags` is described as api filterable in spec doc. However, it doesn't work currently. Leave a comment here for future check.

**Review by ezilber-akamai — COMMENTED** [review]
`TestAccDataSourceImages_basic_smoke` failed locally for me with the error:

`datasource_test.go:33: Step 1/6 error: Check failed: Check 6/25 error: data.linode_images.foobar: Attribute 'images.0.tags.0' expected "test", got "gen2"`

**Comment by yec-akamai** [comment]
> `TestAccDataSourceImages_basic_smoke` failed locally for me with the error:
> 
> `datasource_test.go:33: Step 1/6 error: Check failed: Check 6/25 error: data.linode_images.foobar: Attribute 'images.0.tags.0' expected "test", got "gen2"`

Nice catch! I switched to test if the tags contains `test`.

**Review by ezilber-akamai — APPROVED** [review]
All tests passing locally now. Nice work!

File: linode/image/resource_test.go (hunk 2 above)
Comment (zliang-akamai) [reply linode/image/resource_test.go:219]: I would recommend using `t.Skip("some reasons")` rather than comment it out.

**Review by zliang-akamai — APPROVED** [review]
Tests worked well, and overall looks great! Nice work!

**Pull request merged**